public/jira/rollup/percent-complete/percent-complete.ts

```diff
@@ -0,0 +1,184 @@
+  IssueOrRelease,
+  isDerivedIssue,
+} from "../rollup";
+
+// const BASE_HIERARCHY_LEVEL = 1;
+
+export type PercentCompleteMeta = {
+  /** how many children on average */
+  childCounts: number[];
+  totalDays: number;
+  /** an array of the total of the number of days of work. Used to calculate the average */
+  totalDaysOfWorkForAverage: number[];
+  /** which items need their average set after the average is calculated */
+  needsAverageSet: PercentCompleteRollup[];
+  /** this will be set later */
+  averageTotalDays: number;
+  averageChildCount: number;
+};
+
+export type PercentCompleteRollup = {
+  userSpecifiedValues: boolean;
+  totalWorkingDays: number;
+  completedWorkingDays: number;
+  readonly remainingWorkingDays: number;
+};
+
+export type WithPercentComplete = {
+  completionRollup: PercentCompleteRollup;
+};
+
+const methods = {
+  childrenFirstThenParent,
+};
+export type PercentCompleteCalculations = keyof typeof methods;
+
+function sum(arr: number[]) {
+  return arr.reduce((partialSum, a) => partialSum + a, 0);
+}
+function average(arr: number[]) {
+  return arr.length > 0 ? sum(arr) / arr.length : 0;
+}
+
+export function addPercentComplete<T>(
+  issuesOrReleases: IssueOrRelease<T>[],
+  rollupTimingLevelsAndCalculations: RollupLevelAndCalculation[]
+): IssueOrRelease<T & WithPercentComplete>[] {
+  const groupedIssues = groupIssuesByHierarchyLevelOrType(
+    issuesOrReleases,
+    rollupTimingLevelsAndCalculations
+  );
+
+  // const rollupMethods = rollupTimingLevelsAndCalculations
+  //   .map((rollupData) => rollupData.calculation)
+  //   .reverse();
+  const rolledUpDates = rollupPercentComplete(groupedIssues);
+  const zipped = zipRollupDataOntoGroupedData(groupedIssues, rolledUpDates, (item, values) => ({
+    ...item,
+    completionRollup: values,
+  }));
+  return zipped.flat();
+}
+
+/**
+ *
+ * @param issuesOrReleases Starting from low to high
+ * @param _methodNames
+ * @return
+ */
+export function rollupPercentComplete<T>(groupedHierarchy: IssueOrRelease<T>[][]) {
+  return rollupGroupedHierarchy(groupedHierarchy, {
+    // have to specify the return type on this func so it knows what type the arrays are
+    createMetadataForHierarchyLevel(): PercentCompleteMeta {
+      return {
+        // how many children on average
+        childCounts: [],
+        totalDays: 0,
+
+        // an array of the total of the number of days of work. Used to calculate the average
+        totalDaysOfWorkForAverage: [],
+        // which items need their average set after the average is calculated
+        needsAverageSet: [],
+        // this will be set later
+        averageTotalDays: 0,
+        averageChildCount: 0,
+        // issues: [],
+      };
+    },
+    createRollupDataFromParentAndChild(
+      issueOrRelease,
+      children: PercentCompleteRollup[],
+      hierarchyLevel,
+      metadata
+    ) {
+      const methodName = /*methodNames[hierarchyLevel] ||*/ "childrenFirstThenParent";
```

Comment:
can remove the comment since methodNames is no longer passed to the function

Reply:
a lot of the commented code was in the original and some, like this one, are only meant to be temporarily disabled. Normally i'd say we can just get it from the history if we need it, but I this case it could be difficult to track down in the history. Personally I think i'd still be ok with removing it cuz none of it really seems like something we couldn't come up with again without too much trouble, but i'd like to double check with Justin that there's nothing commented out that he actually needs for later.